### PR TITLE
fix: GET_PATH precedence for JSONExtractScalar

### PR DESCRIPTION
**Pull request by seruman**
I've encountered another precedence issue with json extract scalar this time;

```sql
select JSON('{"K1": "a", "K2": "b"}') as col, CASE WHEN COL ->> '$.K1' = 'a' AND COL ->> '$.K2' = 'b' THEN 'yes' END as r;
-- Error: Conversion Error: Failed to cast value to numerical: {"K1":"a","K2":"b"}

select JSON('{"K1": "a", "K2": "b"}') as col, CASE WHEN (COL ->> '$.K1') = 'a' AND (COL ->> '$.K2') = 'b' THEN 'yes' END as r;
-- ┌─────────────────────┬─────────┐
-- │         col         │    r    │
-- │        json         │ varchar │
-- ├─────────────────────┼─────────┤
-- │ {"K1":"a","K2":"b"} │ yes     │
-- └─────────────────────┴─────────┘

```

### CODE REVIEW

File: tests/test_fakes.py [hunk 1]
```diff
@@ -707,6 +707,15 @@ def test_get_path_precedence(cur: snowflake.connector.cursor.SnowflakeCursor):
     cur.execute("select {'K1': {'K2': 1}} as col where col:K1:K2 > 0")
     assert indent(cur.fetchall()) == [('{\n  "K1": {\n    "K2": 1\n  }\n}',)]
 
+    cur.execute("select {'K1': {'K2': 1}} as col where col:K1:K2 > 0")
+    assert indent(cur.fetchall()) == [('{\n  "K1": {\n    "K2": 1\n  }\n}',)]
+
+    cur.execute("create table example (col variant)")
+
+    cur.execute("""insert into example values ('{"K1": "a", "K2": "b"}')""")
```
Comment (tekumara) [inline tests/test_fakes.py:715]: This gives me the following on the real Snowflake:
> ProgrammingError: 002023 (22000): SQL compilation error:
Expression type does not match column data type, expecting VARIANT but got VARCHAR(22) for column COL

Comment (seruman) [reply tests/test_fakes.py]: Weirdly I've made mistakes moving queries from Snowflake worksheet to tests. 

What is your strategy on testing against real Snowflake?

Comment (seruman) [reply tests/test_fakes.py]: Btw, tests fails due to sqlalchemty, guessing something to do with `snowflake-sqlalchemy` ?

Comment (seruman) [reply tests/test_fakes.py]: Not sure which versions get installed on cached venv previously, but on my local machine it was;
```
snowflake-sqlalchemy==1.5.1
SQLAlchemy==1.4.52
```

on failing run;

```
snowflake-sqlalchemy==1.5.2
SQLAlchemy==2.0.29
```

Comment (tekumara) [reply tests/test_fakes.py]: > What is your strategy on testing against real Snowflake?

I copy-paste the test into [snowflake.ipynb](https://github.com/tekumara/fakesnow/blob/main/notebooks/snowflake.ipynb) and run it there.

Comment (tekumara) [reply tests/test_fakes.py]: Looks like snowflake-sqlalchemy unpinned SQLAlchemy but its now fixed in [1.5.3](https://github.com/snowflakedb/snowflake-sqlalchemy/releases/tag/v1.5.3). So I reran the GHA workflow and its now passed.

**Comment by tekumara** [comment]
Thank you for the fix!

**Comment by seruman** [comment]
> Thank you for the fix!

thanks for this awesome project 🎉

**Pull request merged**